sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py

```diff
@@ -84,14 +122,47 @@ class EventHubProducerClient(ClientBase):
 
     """
 
+    @overload
+    def __init__(
+        self,
+        fully_qualified_namespace: str,
+        eventhub_name: str,
+        credential: "CredentialTypes",
+        *,
+        buffered_mode: Literal[False] = False,
+        **kwargs: Any
+    ) -> None:
+        ...
+
+    @overload
+    def __init__(
+        self,
+        fully_qualified_namespace: str,
+        eventhub_name: str,
+        credential: "CredentialTypes",
+        *,
+        buffered_mode: Literal[True],
+        on_error: Callable[[SendEventTypes, Optional[str], Exception], None],
+        on_success: Callable[[SendEventTypes, Optional[str]], None],
+        max_buffer_length: int = 1500,
+        max_wait_time: float = 1,
```

Comment:
These are documented as `Optional` - i.e. `Union[int, None]`. What would happen if I passed None in? Or is that invalid in a buffered scenario?

Reply:
hmm nevermind - I see the code below means that None would indeed be invalid.

Reply:
if max_wait_time is passed as None, then it defaults to 1 in the buffered mode.